### ISSUE TITLE
Only set object slug if not set

### ISSUE THREAD

**Issue by solverat** [coreshop/CoreShop#2198]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no

https://github.com/coreshop/CoreShop/blob/3b0eccb3d1f427ce4a9526a4ae3588796dc6459d/src/CoreShop/Component/Pimcore/Slug/DataObjectSlugGenerator.php#L81

If the slug has been set once, it never should get touched again from the `DataObjectSlugGenerator`. Currently, the slug gets updated and re-adjusted, if the category/product name has changed.

There are two reasons, why this should be done:
- SEO: If the content manager changes the product name, the URL should not (necessarily) change
- The content manager may want to change the slug to a complete different slug

TL;DR: Setting the slug automatically is ok - but only if the (localized) slug field is empty.

**Comment by dpfaffenbauer** [comment]
SEO: Pimcore automatically creates a redirect for changed slugs, so the URL is still available.

**Comment by solverat** [comment]
Yes, I'm aware of that. But every redirect counts as a negative impact and should be prevented as much as possible.

**Comment by solverat** [comment]
@dpfaffenbauer Yes!! 💪
